src/cgame/cg_rocket_dataformatter.cpp

```diff
@@ -62,22 +62,31 @@ static const char *DisplayAspectString( int w, int h )
 		h = 10;
 	}
 
-	return va( "%d:%d", w, h );
+	return Str::Format( "%d:%d", w, h );
 }
 
+// see resolution_t
 static void CG_Rocket_DFResolution( int handle, const char *data )
 {
 	int w = atoi( Info_ValueForKey(data, "1" ) );
 	int h = atoi( Info_ValueForKey(data, "2" ) );
 
-	if ( w == -1 || h == -1 )
+	if ( w < 0 )
 	{
-		Rocket_DataFormatterFormattedData( handle, "Custom", false );
-		return;
+		std::string displayString = Str::Format( _( "Custom: %dx%d" ), -w, -h );
+		Rocket_DataFormatterFormattedData( handle, displayString.c_str(), false);
+	}
+	else if ( w == 0 )
+	{
+		std::string displayString = Str::Format( _( "Same as screen (%dx%d)" ),
+			cgs.glconfig.vidWidth, cgs.glconfig.vidHeight );
```

Comment:
On my end this displays the current resolution of the game, not the native resolution of the screen.

Reply:
I don't think there is anything we can do about this? We can only get what SDL gives us.

Reply:
By native resolution on the screen, I mean what SDL gives us.

Right now it displays the current resolution of the game, not the screen one.

If I set my screen to `3840×2160` and Unvanquished to windowed `1920×1080`, this line displays “`Same as screen (1920×1080)`”, which is wrong.

Reply:
Whoops, could have sworn that was working before... I'll drop the `%dx%d`.

On the bright side, we can remove two `Cvar::GetValue`s since the info is already there.

Maybe for 0.55 we can add `desktopMode` in the glconfig_t.

Reply:
> Maybe for 0.55 we can add `desktopMode` in the glconfig_t.

Yes.